src/Tasks/Microsoft.NET.Build.Tasks/PublishRelativePathToPackPackagePath.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// 1. Publish PublishRelativePath is a relative <strong>file</strong> path. This need to convert to relative path.
+    /// 2. Due to DotnetTools package format, "tools/{TargetFramework}/any/" is needed in addition to the relative path.
+    /// </summary>
+    public sealed class PublishRelativePathToPackPackagePath : TaskBase
+    {
+        [Required]
+        public ITaskItem[] ResolvedFileToPublish { get; set; }
+
+        [Required]
+        public string PublishDir { get; set; }
+
+        [Required]
+        public string TargetFramework { get; set; }
+
+        [Output]
+        public ITaskItem[] ResolvedFileToPublishWithPackagePath { get; private set; }
+
+        private const char DirectorySeparatorChar = '\\';
+        private const char AltDirectorySeparatorChar = '/';
```

Comment:
These are confusing me since they seem like they might just be `using static Path` when reading the code. These names are generally OS-specific. How about rename to ForwardSeparatorChar and BackwardSeparatorChar. Or WindowsSeparatorChar and UnixSeparatorChar?

Reply:
fixed, renamed to ForwardSeparatorChar and BackwardSeparatorChar